resources/type-system/subtyping.md

```diff
@@ -92,11 +92,7 @@ capture-avoiding substitution of types `T0, ..., Tk` for the type variables
 ## Type equality
 
 We say that a type `T0` is equal to another type `T1` (written `T0 === T1`) if
-the two types are structurally equal up to renaming of bound type variables,
-and equating all top types.
-
-TODO: make these rules explicit.
-
+`T0 <: T1` and `T1 <: T0`.
```

Comment:
cc @stefantsov @johnniwinther @fishythefish @nshahan @liamappelbe @crelier @scheglov This change makes the subtyping rules for generic methods more forgiving.  It is specifically intended to allow a generic method with a legacy bound to override one with a nullable or non-nullable bound.  It also has the effect of allowing overrides that use different variants of top types for bounds. It also accounts for ad hoc normalization (e.g. `T Function<T extends int?>()` is specified to be a subtype of `T Function<T extends int??>()` by this change, and hence this validates the normalization of `int??` to `int?`).

Reply:
Is this change gated behind anything, or can we just update the (legacy) subtyping algorithm?

Reply:
Let's just update in place.